### PR TITLE
[WIP] Adds support for additional HTTP verbs PATCH, OPTIONS and HEAD

### PR DESCRIPTION
**Pull request by ewilde**
# TODO
- [x] Manual testing
- [ ] Wait for gateway release
- [ ] Wait for swarm release
- [ ] Wait for faas-netes release

## Description
Adds support for additional HTTP verbs PATCH, OPTIONS and HEAD

All verbs used with REST APIs are now supported: "GET", "POST", "PUT",
"PATCH", "DELETE", "OPTIONS", "HEAD"

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
Using a custom build of `faas-gateway`

Note in the output below `Http_Method=PATCH`
```
$ echo | ./faas-cli invoke env-test --method PATCH
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=9fcfbb9b0fd2
fprocess=env
HOME=/root
Http_User_Agent=Go-http-client/1.1
Http_Accept_Encoding=gzip
Http_Content_Type=text/plain
Http_X_Call_Id=462b953a-2cf9-4b03-aa16-b7ff0b05c8ed
Http_X_Forwarded_For=10.255.0.2:37292
Http_X_Start_Time=1534620562357807614
Http_Authorization=Basic YWRtaW46YTZlYzIzNGI1MTM3ZGY0NTU2MGFjNGQ3MTNiNGQyZmFiYmRhOTVhMTQyZWRkM2UwMzc0NmUzOGVmZmUyZjY1ZQ==
Http_Method=PATCH
Http_ContentLength=-1
Http_Path=/function/env-test
Http_Host=127.0.0.1:8080
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: proxy/invoke.go [hunk 1]
```diff
@@ -147,7 +147,10 @@ func parseHeaders(headers []string) (map[string]string, error) {
 
 // validateMethod validates the HTTP request method
 func validateHTTPMethod(httpMethod string) error {
-	var allowedMethods = []string{http.MethodGet, http.MethodPost}
+	var allowedMethods = []string{
+		http.MethodGet, http.MethodPost, http.MethodPut, http.MethodPatch, http.MethodDelete,
```
Comment (alexellis) [inline proxy/invoke.go]: SGTM - let's just remove `Head` and `Options` and I'll merge for milestone: `0.7.2`.

Comment (ewilde) [reply proxy/invoke.go]: done :+1:

**Comment by alexellis** [comment]
Please can you set the milestone to 0.7.2?

**Comment by ewilde** [comment]
Derek set milestone: 0.7.2

**Pull request merged**